src/System.Buffers/src/System/Buffers/DefaultArrayPool.cs

```diff
@@ -49,6 +60,7 @@ public override T[] Rent(int minimumLength)
                     // If the bucket has an array left and returned it, give it to the caller
                     if (buffer != null)
                     {
+                        _eventSource.BufferRented(buffer.GetHashCode(), buffer.Length, _buckets[i].GetHashCode());
```

Comment:
Since we're concerned about minimizing as much overhead as possible, you might do:
```C#
var log = ArrayPoolEventSource.Log;
if (log.IsEnabled()
{
    log.BufferRented(buffer.GetHashCode(), buffer.Length, _buckets[i].GetHashCode();
}
```
such that we're only computing hash codes if we're actually going to be logging.